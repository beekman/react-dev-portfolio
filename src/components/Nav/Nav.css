@value colors: '../colors.css';
@value lightblue, yellow, offwhite, white, darkgray, lightyellow, whiteblue from colors;

.Header{
  position: fixed;
  z-index: 999999999;
}

.Header{
  flex-wrap: wrap;
  align-items: flex-start;
  position: fixed;
  background-color: RGBA(0,0,0,1);
  width: 100%;

  h1{
    display: inline-block;
  }

  .Logo {
    font-family: 'Play', sans-serif;
    line-height: 1em;
    color: white;
    margin: 0;
    padding: 0.5rem 0.75rem;
    display: inline-block;

    strong {
    font-size: 2rem;
    font-weight: bold;
    font-style: normal;
    position: relative;
    display: inline-block;
    transition-timing-function: ease;
    transition-duration: 1s;
    transition-property: color;
    text-transform: lowercase;
    color: #37e8fd;

    &:hover {
      color: #d7ffff;
    }
  }

    > b {
    font-size: 2rem;
    font-weight: 700;
    position: relative;
    z-index: 10;
    transition-timing-function: ease;
    transition-duration: .7s;
    transition-property: color;
    color: #bafd31;
    }

    > b:hover {
      color: #d7ffff;
    }
    .dev {
      font-size: 16px;
      color: offwhite;
    }
  }

  .Nav{
    display: inline-block;

    ul{
      display: flex;
      flex-wrap: wrap;
      width: 100%;
      list-style-type: none;
      justify-content: center;

      li {
        color: #FFFFFF;
        background: #232323;

        a {
          text-decoration: none;
          color: lightblue;
          padding: 0 1rem;

          &:active, &:hover {
            color: whiteblue;
            text-shadow: 0 0 5px yellow, 0 0 10px yellow, 0 0 15px yellow, 0 0 20px yellow, 0 0 30px yellow, 0 0 40px yellow, 0 0 55px yellow, 0 0 75px yellow;
            cursor: pointer;
          }

          &.active {
            color: lightyellow;
            background-color: darkgray;
            text-shadow: 0 -1px 4px whiteblue, 0 -2px 10px lightyellow, 0 -10px 20px yellow;
          }
        }
      }
    }
  }
}
