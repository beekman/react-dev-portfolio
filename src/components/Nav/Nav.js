import React from 'react';
import styles from '../App';
import { Link } from 'react-scroll';
export default function Nav() {
  return (
    <nav className={styles.Nav}>
      <ul>
        <Link
          to='About'
          spy={true}
          smooth={true}
          duration={600}
          className={styles.Link}
          activeClass={styles.Active}>
          <li>About</li>
        </Link>
        <Link
          to='Skills'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={600}
          className={styles.Link}
          activeClass={styles.Active}>
          <li>Skills</li>
        </Link>
        <Link
          to='Projects'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={600}
          className={styles.Link}
          activeClass={styles.Active}>
          <li>Projects</li>
        </Link>
        <Link
          to='Contact'
          spy={true}
          hashSpy={true}
          smooth={true}
          duration={600}
          className={styles.Link}
          activeClass={styles.Active}>
          <li>Contact</li>
        </Link>
      </ul>
    </nav>
  );
}
