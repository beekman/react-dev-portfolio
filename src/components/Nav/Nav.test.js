import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';

describe('Nav component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper).toMatchSnapshot();
  });
});
