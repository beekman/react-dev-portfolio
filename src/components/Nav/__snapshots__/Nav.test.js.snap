// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<header
  className="Header"
>
  <h1
    className="Logo"
  >
    <strong>
      Ben
    </strong>
    <b>
      Beekman
    </b>
    <span
      className="dev"
    >
      .dev
    </span>
  </h1>
  <nav
    className="Nav"
  >
    <ul>
      <li>
        <Link
          activeClass="active"
          className="link"
          duration={600}
          offset={-200}
          smooth={true}
          spy={true}
          to="About"
        >
          About
        </Link>
      </li>
      <li>
        <Link
          activeClass="active"
          className="link"
          duration={600}
          hashSpy={true}
          offset={0}
          smooth={true}
          spy={true}
          to="Skills"
        >
          Skills
        </Link>
      </li>
      <li>
        <Link
          activeClass="active"
          className="link"
          duration={600}
          hashSpy={true}
          offset={0}
          smooth={true}
          spy={true}
          to="Projects"
        >
          Projects
        </Link>
      </li>
      <li>
        <Link
          activeClass="active"
          className="link"
          duration={400}
          hashSpy={true}
          offset={0}
          smooth={true}
          spy={true}
          to="Contact"
        >
          Connect
        </Link>
      </li>
    </ul>
  </nav>
</header>
`;
